### PR TITLE
Move `CustomerInfo` to non-expect classes and use new mappings and models modules

### PR DESCRIPTION
**Pull request by tonidero**
### Description

This PR moves the `CustomerInfo`, `EntitlementInfos`, `EntitlementInfo` and `Transaction` models to the `models` module, converts them to classes without expect/actual and moves their mapping to the `mappings` module.

### CODE REVIEW

File: revenuecatui/src/androidMain/kotlin/com/revenuecat/purchases/kmp/ui/revenuecatui/PaywallOptionsKtx.kt [hunk 1]
```diff
@@ -1,8 +1,9 @@
 package com.revenuecat.purchases.kmp.ui.revenuecatui
 
-import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.CustomerInfo as AndroidCustomerInfo
 import com.revenuecat.purchases.Package
 import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.kmp.mappings.toCustomerInfo
```
Comment (tonidero) [inline revenuecatui/src/androidMain/kotlin/com/revenuecat/purchases/kmp/ui/revenuecatui/PaywallOptionsKtx.kt:6]: Such a big refactor to be able to access this 😅

**Review by JayShortway — APPROVED** [review]
LGTM! Seems like the modules are working for now! Just a single nit, spread out over a few comments haha.

File: mappings/src/iosMain/kotlin/com/revenuecat/purchases/kmp/mappings/ktx/Map.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.revenuecat.purchases.kmp.mappings.ktx
+
+internal inline fun <K1, V1, K2, V2> Map<out K1, V1>.mapEntries(
+    transform: (Map.Entry<K1, V1>) -> Pair<K2, V2>
+): Map<K2, V2> = entries.associate(transform)
```
[the remark this answers is not in the document]
Comment (tonidero) [reply mappings/src/iosMain/kotlin/com/revenuecat/purchases/kmp/mappings/ktx/Map.kt:5]: This is actually copied over from the `core` module. Once we move all the mappings, we should be able to delete the one from there.

[the remark this answers is not in the document]
Comment (JayShortway) [reply mappings/src/iosMain/kotlin/com/revenuecat/purchases/kmp/mappings/ktx/Map.kt:5]: Oh, did I come up with this then? 😅 
image

**Review by JayShortway — APPROVED** [review]
Good stuff!

File: build.gradle.kts [hunk 3]
```diff
@@ -100,7 +100,7 @@ allprojects {
 }
 
 apiValidation {
-    ignoredProjects.addAll(listOf("apiTester", "composeApp"))
+    ignoredProjects.addAll(listOf("apiTester", "composeApp", "mappings"))
```
Comment (tonidero) [inline build.gradle.kts:103]: Removed the `mappings` module from here, that shouldn't be public.

**Comment by tonidero** [comment]
Tests are not passing in these branches it seems (only on PRs pointing to main). Tests are passing locally for me now though, so going to merge this.

**Pull request merged**